crates/collab/src/api/events.rs

```diff
@@ -495,173 +495,7 @@ fn for_snowflake(
-                    }
-                    "welcome page: toggle vim" => "Welcome Vim Mode Toggled".to_string(),
-                    "welcome page: view docs" => "Welcome Documentation Viewed".to_string(),
-
-                    // Extensions
-                    "extensions page: open" => "Extensions Page Opened".to_string(),
-                    "extensions: install extension" => "Extension Installed".to_string(),
-                    "extensions: uninstall extension" => "Extension Uninstalled".to_string(),
-
-                    // Misc
-                    "markdown preview: open" => "Markdown Preview Opened".to_string(),
-                    "project diagnostics: open" => "Project Diagnostics Opened".to_string(),
-                    "project search: open" => "Project Search Opened".to_string(),
-                    "repl sessions: open" => "REPL Session Started".to_string(),
-
-                    // Feature Upsell
-                    "feature upsell: toggle vim" => {
-                        properties["source"] = json!("Feature Upsell");
-                        "Vim Mode Toggled".to_string()
-                    }
-                    _ => e
-                        .operation
-                        .strip_prefix("feature upsell: viewed docs (")
-                        .and_then(|s| s.strip_suffix(')'))
-                        .map_or_else(
-                            || format!("Unknown App Event: {}", e.operation),
-                            |docs_url| {
-                                properties["url"] = json!(docs_url);
-                                properties["source"] = json!("Feature Upsell");
-                                "Documentation Viewed".to_string()
-                            },
-                        ),
-                };
-                (event_type, properties)
-            }
-            Event::Setting(e) => (
-                "Settings Changed".to_string(),
-                serde_json::to_value(e).unwrap(),
-            ),
-            Event::Extension(e) => (
-                "Extension Loaded".to_string(),
-                serde_json::to_value(e).unwrap(),
-            ),
-            Event::Edit(e) => (
-                "Editor Edited".to_string(),
-                serde_json::to_value(e).unwrap(),
-            ),
-            Event::Action(e) => (
-                "Action Invoked".to_string(),
-                serde_json::to_value(e).unwrap(),
-            ),
-            Event::Repl(e) => (
-                "Kernel Status Changed".to_string(),
-                serde_json::to_value(e).unwrap(),
-            ),
-            Event::Flexible(e) => (
-                e.event_type.clone(),
-                serde_json::to_value(&e.event_properties).unwrap(),
-            ),
-        };
-
+        let mut event_properties = serde_json::to_value(&event.event.event_properties).unwrap();
         if let serde_json::Value::Object(ref mut map) = event_properties {
```

Comment:
I'm assuming events in the old format will get rejected here?

Reply:
At some point, users will be upgraded to a version where all events are generated via `event!` and those events no longer interact with the code I deleted in collab, as it was only meant to handle the old historical event format. Do I need to roll back the change that flattens the FlexbileEvent out then? I only removed that as it would be the only variable left in the event enum.

Reply:
I think so (unfortunately) as otherwise there'll be a hard cut-off date. If you could have collab continue parsing either format that would also work.

Reply:
Ok, I rolled back to `FlexibleEvent`. The idea of having collab parse the wrapped Event and the event data flattened out is appealing, and then we could just remove the wrapped one later down the line, but not sure how to deal with that now. I think this should be good to land after a few more releases. Really just wanted that migration code out of collab.